dfx-library/dfxplugin-audiounit.cpp

```diff
@@ -1577,8 +1577,9 @@ OSStatus DfxPlugin::ChangeStreamFormat(AudioUnitScope inScope, AudioUnitElement
 void DfxPlugin::UpdateInPlaceProcessingState()
 {
 #if !TARGET_PLUGIN_IS_INSTRUMENT
-	// we can't do in-place audio rendering if there are different numbers of audio inputs and outputs
-	if (getnuminputs() != getnumoutputs())
+	// we can't do in-place audio rendering if there are different numbers of audio inputs and outputs 
+	// or the audio rendering is summed into the output
+	if ((getnuminputs() != getnumoutputs()) || mAudioProcessingAccumulatingOnly)
```

Comment:
I guess I can only comment on delta lines...

I think it would make sense to have a line or two of documentation on setAudioProcessingMustAccumulate in dfxplugin.h since it's part of the client interface. I think accumulate means to output with += instead of =. Does calling this function (with true) just declare "+= is what I do in processaudio()" or is it saying "I am only capable of += so better use me that way"? The latter is what the name suggests to me but I can't remember/see how a plugin could be informed to use accumulation or not.

Reply:
It's the latter, and yes, especially given that the idea of accumulating processing now does not exist anywhere else, I definitely should document!

Reply:
Talking/typing this out also made me realize that the language of "accumulating" just doesn't make sense anymore.  So in addition to documenting this, I flipped the language; it is expressed now in terms of whether the plugin support in-place processing, rather than whether it requires "accumulating" processing (which wasn't even accurate given that we internally implement with zeroed output buffers, and hey look, the language even led to me getting myself mixed up enough to lose track of the fact that I had to zero those output buffers).